### PR TITLE
GH-111062: Apply reusable workflow for the Linux CI

### PR DESCRIPTION
**Pull request by NCLI**
* Issue: gh-111062

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -332,6 +332,75 @@ jobs:
...
+    - name: Build CPython out-of-tree
+      working-directory: ${{ env.CPYTHON_BUILDDIR }}
+      run: make -j4
+    - name: Display build info
+      working-directory: ${{ env.CPYTHON_BUILDDIR }}
+      run: make pythoninfo
+    - name: Remount sources writable for tests
+      # some tests write to srcdir, lack of pyc files slows down testing
+      run: sudo mount $CPYTHON_RO_SRCDIR -oremount,rw
+    - name: Tests
```
Comment (corona10) [inline .github/workflows/build.yml]: FYI, we don't need to run test code at this moment :)

**Comment by dimaqq** [comment]
Is this superceded by #111446 now?

**Comment by dimaqq** [comment]
"nogil" -> "free-threaded"
let's build this on top of #111446

**Comment by corona10** [comment]
Duplicated with https://github.com/python/cpython/pull/111452

**Pull request closed without merging**